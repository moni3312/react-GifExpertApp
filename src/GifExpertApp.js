import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = ({defaultCategories = []})=>{

const [categories, setCategories] = useState(defaultCategories)


return  <>
            <h2 className= "titulo">GifExpertApp</h2>
            <br />
            <AddCategory setCategories={setCategories} />
            <hr />
            <nl>
                {   
                    categories.map(category =>
                        <GifGrid
                        key= {category} 
                        category = {category} 
                        />
                    )
                }
            </nl>
        </>

}; 
export default GifExpertApp;