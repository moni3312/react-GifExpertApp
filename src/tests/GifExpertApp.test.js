import {shallow} from 'enzyme';
import GifExpertApp from "../GifExpertApp";

describe('Pruebas con <GifExpertApp />', () =>{

    test('debe de hacer match con el snapshot', ()=>{
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
 
    });

    test('debe de mostras una lista de categorias', ()=>{
        const categories = ['One Punch', 'Dragon Ball Z'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
 
    });

});