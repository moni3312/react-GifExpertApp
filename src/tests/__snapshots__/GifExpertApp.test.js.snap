// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h2
    className="titulo"
  >
    GifExpertApp
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <nl />
</Fragment>
`;

exports[`Pruebas con <GifExpertApp /> debe de mostras una lista de categorias 1`] = `
<Fragment>
  <h2
    className="titulo"
  >
    GifExpertApp
  </h2>
  <br />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <nl>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </nl>
</Fragment>
`;
